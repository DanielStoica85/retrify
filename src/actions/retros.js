import uuid from 'uuid';

// Add retro
export const addRetro = ({
    title = '',
    description = '',
    createdAt = 0
} = {}) => ({
    type: 'ADD_RETRO',
    retro: {
        id: uuid(),
        title,
        description,
        createdAt
    }
});

// Remove retro
export const removeRetro = id => ({
    type: 'REMOVE_RETRO',
    id
});

// Edit retro
export const editRetro = (id, updates) => ({
    type: 'EDIT_RETRO',
    id,
    updates
});
