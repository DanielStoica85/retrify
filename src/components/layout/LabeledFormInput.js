import React from 'react';
import { FormGroup, Label, Input } from 'reactstrap';

const LabeledFormInput = props => {
    const {
        labelText,
        labelFor,
        inputType,
        inputName,
        inputId,
        inputPlaceholder,
        handleChange,
        value
    } = props;
    return (
        <FormGroup>
            <Label for={labelFor}>{labelText}</Label>
            <Input
                type={inputType}
                name={inputName}
                id={inputId}
                placeholder={inputPlaceholder}
                onChange={handleChange}
                value={value}
            />
        </FormGroup>
    );
};

export default LabeledFormInput;
