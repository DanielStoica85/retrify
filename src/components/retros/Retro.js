import React from 'react';
import {
    Card,
    CardBody,
    CardTitle,
    CardSubtitle,
    CardFooter
} from 'reactstrap';
import PropTypes from 'prop-types';

const Retro = props => {
    const { title, author, description, createdAt } = props.retro;
    return (
        <div>
            <Card className="mt-3">
                <CardBody>
                    <CardTitle>{title}</CardTitle>
                    <CardSubtitle> {description}</CardSubtitle>
                </CardBody>
                <CardFooter
                    style={{
                        backgroundColor: 'lightgray',
                        borderColor: 'lightgray'
                    }}
                >
                    Added by {author} on {createdAt}
                </CardFooter>
            </Card>
        </div>
    );
};

Retro.propTypes = {
    retro: PropTypes.object
};

export default Retro;
