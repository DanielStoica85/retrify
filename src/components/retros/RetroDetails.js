import React from 'react';
import PropTypes from 'prop-types';

const RetroDetails = props => {
    const id = props.match.params.id;
    return (
        <div className="container section project-details">
            <div className="card">
                <div className="card-content">
                    <span className="card-title">Retro Title #{id} </span>
                    <p>
                        Lorem ipsum dolor sit amet consectetur adipisicing elit.
                        Perspiciatis velit ab optio eius corporis iste facilis,
                        sit eum distinctio inventore repudiandae obcaecati
                        eveniet sunt, quisquam iure quaerat amet reiciendis
                        error?
                    </p>
                </div>
                <div className="card-action grey lighten-4 grey-text">
                    <div>Posted by Daniel</div>
                    <div>May 13th, 2am</div>
                </div>
            </div>
        </div>
    );
};

RetroDetails.propTypes = {
    match: PropTypes.object
};

export default RetroDetails;
