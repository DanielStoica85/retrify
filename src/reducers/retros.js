// Expenses reducer
const retrosReducerDefaultState = [];

export default (state = retrosReducerDefaultState, action) => {
    switch (action.type) {
        case 'ADD_RETRO':
            return [...state, action.retro];
        case 'EDIT_RETRO':
            return state.map(retro => {
                if (retro.id === action.id) {
                    return {
                        ...retro,
                        ...action.updates
                    };
                } else {
                    return retro;
                }
            });
        case 'REMOVE_RETRO':
            return state.filter(retro => retro.id !== action.id);
        case 'SET_RETROS':
            return action.retros;
        default:
            return state;
    }
};
