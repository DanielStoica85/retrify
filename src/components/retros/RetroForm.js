import React, { Component } from 'react';
import LabeledFormInput from '../layout/LabeledFormInput';
import LabeledFormSelect from '../layout/LabeledFormSelect';
import { Button, Form, Label, FormGroup, Alert } from 'reactstrap';

import moment from 'moment';
import 'react-dates/initialize';
import { SingleDatePicker } from 'react-dates';
import 'react-dates/lib/css/_datepicker.css';

class RetroForm extends Component {
    constructor(props) {
        super(props);
        this.state = {
            title: props.retro ? props.retro.title : '',
            description: props.retro ? props.retro.description : '',
            author: props.retro ? props.retro.author : '',
            type: props.retro ? props.retro.type : '',
            createdAt: props.retro ? moment(props.retro.createdAt) : moment(),
            calendarFocused: false,
            error: ''
        };
    }

    handleChange = e => {
        this.setState({
            [e.target.id]: e.target.value
        });
    };

    handleTypeChange = e => {
        const type = e.target.value;
        if (type !== 'Select') {
            this.setState({ type });
        } else {
            this.setState({ type: '' });
        }
    };

    handleDateChange = createdAt => {
        if (createdAt) {
            this.setState(() => ({ createdAt }));
        }
    };

    handleFocusChange = ({ focused }) => {
        this.setState(() => ({ calendarFocused: focused }));
    };

    handleSubmit = e => {
        e.preventDefault();
        if (
            !this.state.title ||
            !this.state.author ||
            !this.state.createdAt ||
            !this.state.type
        ) {
            this.setState(() => ({
                error:
                    'Please provide a title, author name, retro type and a date for this retro.'
            }));
        } else {
            this.setState(() => ({ error: '' }));
            this.props.onSubmit({
                title: this.state.title,
                author: this.state.author,
                description: this.state.description,
                type: this.state.type,
                createdAt: this.state.createdAt.valueOf()
            });
        }
    };

    render() {
        return (
            <Form onSubmit={this.handleSubmit}>
                {this.state.error && (
                    <Alert color="danger">{this.state.error}</Alert>
                )}
                <LabeledFormInput
                    labelText="Retro Board Name*"
                    labelFor="title"
                    inputType="text"
                    inputName="title"
                    inputId="title"
                    value={this.state.title}
                    handleChange={this.handleChange}
                />
                <LabeledFormInput
                    labelText="Author Name*"
                    labelFor="author"
                    inputType="text"
                    inputName="author"
                    inputId="author"
                    value={this.state.author}
                    handleChange={this.handleChange}
                />
                <LabeledFormInput
                    labelText="Description"
                    labelFor="description"
                    inputType="textarea"
                    inputName="description"
                    inputId="description"
                    inputPlaceHolder="Give your board an (optional) description..."
                    value={this.state.description}
                    handleChange={this.handleChange}
                />
                <LabeledFormSelect
                    labelText="Type of Retrospective*"
                    labelFor="type"
                    inputType="select"
                    inputName="type"
                    inputId="type"
                    value={this.state.type}
                    options={[
                        'Select',
                        'What went well / What did not go well',
                        'Start / Stop / Continue'
                    ]}
                    handleChange={this.handleTypeChange}
                />
                <FormGroup>
                    <Label for="date-picker">Date*</Label>
                    <br />
                    <SingleDatePicker
                        date={this.state.createdAt}
                        onDateChange={this.handleDateChange}
                        focused={this.state.calendarFocused}
                        onFocusChange={this.handleFocusChange}
                        numberOfMonths={1}
                        id="date-picker"
                    />
                </FormGroup>
                <FormGroup>
                    <Button color="primary">Save Retro</Button>
                </FormGroup>
            </Form>
        );
    }
}

export default RetroForm;
