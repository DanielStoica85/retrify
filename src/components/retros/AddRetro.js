import React from 'react';
import { connect } from 'react-redux';
import { addRetro } from '../../actions/retros';
import { Container } from 'reactstrap';
import RetroForm from './RetroForm';

export class AddRetro extends React.Component {
    addRetro = retro => {
        this.props.addRetro(retro);
        this.props.history.push('/');
    };

    render() {
        return (
            <Container>
                <h4 className="grey-text text-darken-3">Add Retro Board</h4>
                <RetroForm onSubmit={this.addRetro} />
            </Container>
        );
    }
}

const mapDispatchToProps = dispatch => {
    return {
        addRetro: retro => dispatch(addRetro(retro))
    };
};

export default connect(
    undefined,
    mapDispatchToProps
)(AddRetro);
