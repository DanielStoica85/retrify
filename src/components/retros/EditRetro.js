import React from 'react';
import { connect } from 'react-redux';
import { startEditRetro } from '../../actions/retros';
import { Container } from 'reactstrap';
import RetroForm from './RetroForm';
import PropTypes from 'prop-types';

export class EditRetro extends React.Component {
    onSubmit = retro => {
        this.props.startEditRetro(this.props.retro.id, retro);
        this.props.history.push('/');
    };
    render() {
        return (
            <Container>
                <h4 className="grey-text text-darken-3">
                    Edit Retro Board with id {this.props.match.params.id}
                </h4>
                <RetroForm retro={this.props.retro} onSubmit={this.onSubmit} />
            </Container>
        );
    }
}

const mapDispatchToProps = (dispatch, props) => {
    return {
        startEditRetro: (id, retro) => dispatch(startEditRetro(id, retro))
    };
};

const mapStateToProps = (state, props) => {
    const id = props.match.params.id;
    return {
        retro: state.retros.find(retro => retro.id.toString() === id)
    };
};

EditRetro.propTypes = {
    startEditRetro: PropTypes.func
};

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(EditRetro);
