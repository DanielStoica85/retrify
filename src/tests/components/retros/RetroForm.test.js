import React from 'react';
import { shallow } from 'enzyme';
import RetroForm from '../../../components/retros/RetroForm';
import LabeledFormInput from '../../../components/layout/LabeledFormInput';
import { Form } from 'reactstrap';
import retros from '../../fixtures/retros';

describe('RetroForm component', () => {
    it('should render correctly', () => {
        const component = shallow(<RetroForm />);
        expect(component).toMatchSnapshot();
    });

    it('should render correctly with expense data', () => {
        const component = shallow(<RetroForm retro={retros[1]} />);
        expect(component).toMatchSnapshot();
    });

    it('should render error for invalid form submission', () => {
        const component = shallow(<RetroForm />);
        component.find(Form).simulate('submit', {
            preventDefault: () => {}
        });
        expect(component.state('error').length).toBeGreaterThan(0);
        expect(component).toMatchSnapshot();
    });

    it('should set retro title on input change', () => {
        const value = 'Retro Title';
        const component = shallow(<RetroForm />);
        component
            .find(LabeledFormInput)
            .at(0)
            .prop('handleChange')({ target: { value } });
        expect(component.state('title')).toBe(value);
    });

    it('should set author name on input change', () => {
        const value = 'Author Name';
        const component = shallow(<RetroForm />);
        component
            .find(LabeledFormInput)
            .at(1)
            .prop('handleChange')({ target: { value } });
        expect(component.state('author')).toBe(value);
    });

    it('should set description on input change', () => {
        const value = 'Description';
        const component = shallow(<RetroForm />);
        component
            .find(LabeledFormInput)
            .at(2)
            .prop('handleChange')({ target: { value } });
        expect(component.state('description')).toBe(value);
    });
});
