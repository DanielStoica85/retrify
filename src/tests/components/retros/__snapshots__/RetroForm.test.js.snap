// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RetroForm component should render correctly 1`] = `
<Form
  onSubmit={[Function]}
  tag="form"
>
  <LabeledFormInput
    handleChange={[Function]}
    inputId="title"
    inputName="title"
    inputType="text"
    labelFor="title"
    labelText="Retro Board Name*"
    value=""
  />
  <LabeledFormInput
    handleChange={[Function]}
    inputId="author"
    inputName="author"
    inputType="text"
    labelFor="author"
    labelText="Author Name*"
    value=""
  />
  <LabeledFormInput
    handleChange={[Function]}
    inputId="description"
    inputName="description"
    inputPlaceHolder="Give your board an (optional) description..."
    inputType="textarea"
    labelFor="description"
    labelText="Description"
    value=""
  />
  <FormGroup
    tag="div"
  >
    <Label
      for="date-picker"
      tag="label"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      Date*
    </Label>
    <br />
    <withStyles(SingleDatePicker)
      anchorDirection="left"
      appendToBody={false}
      block={false}
      calendarInfoPosition="bottom"
      customCloseIcon={null}
      customInputIcon={null}
      date={"1970-01-01T00:00:00.000Z"}
      daySize={39}
      disableScroll={false}
      disabled={false}
      displayFormat={[Function]}
      enableOutsideDays={false}
      firstDayOfWeek={null}
      focused={false}
      hideKeyboardShortcutsPanel={false}
      horizontalMargin={0}
      horizontalMonthPadding={13}
      id="date-picker"
      initialVisibleMonth={null}
      inputIconPosition="before"
      isDayBlocked={[Function]}
      isDayHighlighted={[Function]}
      isOutsideRange={[Function]}
      isRTL={false}
      keepFocusOnInput={false}
      keepOpenOnDateSelect={false}
      monthFormat="MMMM YYYY"
      navNext={null}
      navPrev={null}
      noBorder={false}
      numberOfMonths={1}
      onClose={[Function]}
      onDateChange={[Function]}
      onFocusChange={[Function]}
      onNextMonthClick={[Function]}
      onPrevMonthClick={[Function]}
      openDirection="down"
      orientation="horizontal"
      phrases={
        Object {
          "calendarLabel": "Calendar",
          "chooseAvailableDate": [Function],
          "clearDate": "Clear Date",
          "closeDatePicker": "Close",
          "dateIsSelected": [Function],
          "dateIsUnavailable": [Function],
          "enterKey": "Enter key",
          "escape": "Escape key",
          "hideKeyboardShortcutsPanel": "Close the shortcuts panel.",
          "homeEnd": "Home and end keys",
          "jumpToNextMonth": "Move forward to switch to the next month.",
          "jumpToPrevMonth": "Move backward to switch to the previous month.",
          "keyboardNavigationInstructions": "Press the down arrow key to interact with the calendar and
  select a date. Press the question mark key to get the keyboard shortcuts for changing dates.",
          "keyboardShortcuts": "Keyboard Shortcuts",
          "leftArrowRightArrow": "Right and left arrow keys",
          "moveFocusByOneDay": "Move backward (left) and forward (right) by one day.",
          "moveFocusByOneMonth": "Switch months.",
          "moveFocusByOneWeek": "Move backward (up) and forward (down) by one week.",
          "moveFocustoStartAndEndOfWeek": "Go to the first or last day of a week.",
          "openThisPanel": "Open this panel.",
          "pageUpPageDown": "page up and page down keys",
          "questionMark": "Question mark",
          "returnFocusToInput": "Return to the date input field.",
          "selectFocusedDate": "Select the date in focus.",
          "showKeyboardShortcutsPanel": "Open the keyboard shortcuts panel.",
          "upArrowDownArrow": "up and down arrow keys",
        }
      }
      placeholder="Date"
      readOnly={false}
      regular={false}
      renderCalendarInfo={null}
      renderDayContents={null}
      renderMonthElement={null}
      renderMonthText={null}
      reopenPickerOnClearDate={false}
      required={false}
      screenReaderInputMessage=""
      showClearDate={false}
      showDefaultInputIcon={false}
      small={false}
      verticalHeight={null}
      verticalSpacing={22}
      weekDayFormat="dd"
      withFullScreenPortal={false}
      withPortal={false}
    />
  </FormGroup>
  <FormGroup
    tag="div"
  >
    <Button
      color="primary"
      tag="button"
    >
      Save Retro
    </Button>
  </FormGroup>
</Form>
`;

exports[`RetroForm component should render correctly with expense data 1`] = `
<Form
  onSubmit={[Function]}
  tag="form"
>
  <LabeledFormInput
    handleChange={[Function]}
    inputId="title"
    inputName="title"
    inputType="text"
    labelFor="title"
    labelText="Retro Board Name*"
    value="Second Retro"
  />
  <LabeledFormInput
    handleChange={[Function]}
    inputId="author"
    inputName="author"
    inputType="text"
    labelFor="author"
    labelText="Author Name*"
    value="Costel"
  />
  <LabeledFormInput
    handleChange={[Function]}
    inputId="description"
    inputName="description"
    inputPlaceHolder="Give your board an (optional) description..."
    inputType="textarea"
    labelFor="description"
    labelText="Description"
    value="Lorem ipsum dolor sit amet consectetur adipisicing elit. Deleniti itaque aut eligendi laborum excepturi animi hic similique impedit enim velit."
  />
  <FormGroup
    tag="div"
  >
    <Label
      for="date-picker"
      tag="label"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      Date*
    </Label>
    <br />
    <withStyles(SingleDatePicker)
      anchorDirection="left"
      appendToBody={false}
      block={false}
      calendarInfoPosition="bottom"
      customCloseIcon={null}
      customInputIcon={null}
      date={"2019-07-27T19:32:17.000Z"}
      daySize={39}
      disableScroll={false}
      disabled={false}
      displayFormat={[Function]}
      enableOutsideDays={false}
      firstDayOfWeek={null}
      focused={false}
      hideKeyboardShortcutsPanel={false}
      horizontalMargin={0}
      horizontalMonthPadding={13}
      id="date-picker"
      initialVisibleMonth={null}
      inputIconPosition="before"
      isDayBlocked={[Function]}
      isDayHighlighted={[Function]}
      isOutsideRange={[Function]}
      isRTL={false}
      keepFocusOnInput={false}
      keepOpenOnDateSelect={false}
      monthFormat="MMMM YYYY"
      navNext={null}
      navPrev={null}
      noBorder={false}
      numberOfMonths={1}
      onClose={[Function]}
      onDateChange={[Function]}
      onFocusChange={[Function]}
      onNextMonthClick={[Function]}
      onPrevMonthClick={[Function]}
      openDirection="down"
      orientation="horizontal"
      phrases={
        Object {
          "calendarLabel": "Calendar",
          "chooseAvailableDate": [Function],
          "clearDate": "Clear Date",
          "closeDatePicker": "Close",
          "dateIsSelected": [Function],
          "dateIsUnavailable": [Function],
          "enterKey": "Enter key",
          "escape": "Escape key",
          "hideKeyboardShortcutsPanel": "Close the shortcuts panel.",
          "homeEnd": "Home and end keys",
          "jumpToNextMonth": "Move forward to switch to the next month.",
          "jumpToPrevMonth": "Move backward to switch to the previous month.",
          "keyboardNavigationInstructions": "Press the down arrow key to interact with the calendar and
  select a date. Press the question mark key to get the keyboard shortcuts for changing dates.",
          "keyboardShortcuts": "Keyboard Shortcuts",
          "leftArrowRightArrow": "Right and left arrow keys",
          "moveFocusByOneDay": "Move backward (left) and forward (right) by one day.",
          "moveFocusByOneMonth": "Switch months.",
          "moveFocusByOneWeek": "Move backward (up) and forward (down) by one week.",
          "moveFocustoStartAndEndOfWeek": "Go to the first or last day of a week.",
          "openThisPanel": "Open this panel.",
          "pageUpPageDown": "page up and page down keys",
          "questionMark": "Question mark",
          "returnFocusToInput": "Return to the date input field.",
          "selectFocusedDate": "Select the date in focus.",
          "showKeyboardShortcutsPanel": "Open the keyboard shortcuts panel.",
          "upArrowDownArrow": "up and down arrow keys",
        }
      }
      placeholder="Date"
      readOnly={false}
      regular={false}
      renderCalendarInfo={null}
      renderDayContents={null}
      renderMonthElement={null}
      renderMonthText={null}
      reopenPickerOnClearDate={false}
      required={false}
      screenReaderInputMessage=""
      showClearDate={false}
      showDefaultInputIcon={false}
      small={false}
      verticalHeight={null}
      verticalSpacing={22}
      weekDayFormat="dd"
      withFullScreenPortal={false}
      withPortal={false}
    />
  </FormGroup>
  <FormGroup
    tag="div"
  >
    <Button
      color="primary"
      tag="button"
    >
      Save Retro
    </Button>
  </FormGroup>
</Form>
`;

exports[`RetroForm component should render error for invalid form submission 1`] = `
<Form
  onSubmit={[Function]}
  tag="form"
>
  <Alert
    closeAriaLabel="Close"
    color="danger"
    fade={true}
    isOpen={true}
    tag="div"
    transition={
      Object {
        "appear": true,
        "baseClass": "fade",
        "baseClassActive": "show",
        "enter": true,
        "exit": true,
        "in": true,
        "mountOnEnter": false,
        "onEnter": [Function],
        "onEntered": [Function],
        "onEntering": [Function],
        "onExit": [Function],
        "onExited": [Function],
        "onExiting": [Function],
        "tag": "div",
        "timeout": 150,
        "unmountOnExit": true,
      }
    }
  >
    Please provide a title, author name and a date for this retro.
  </Alert>
  <LabeledFormInput
    handleChange={[Function]}
    inputId="title"
    inputName="title"
    inputType="text"
    labelFor="title"
    labelText="Retro Board Name*"
    value=""
  />
  <LabeledFormInput
    handleChange={[Function]}
    inputId="author"
    inputName="author"
    inputType="text"
    labelFor="author"
    labelText="Author Name*"
    value=""
  />
  <LabeledFormInput
    handleChange={[Function]}
    inputId="description"
    inputName="description"
    inputPlaceHolder="Give your board an (optional) description..."
    inputType="textarea"
    labelFor="description"
    labelText="Description"
    value=""
  />
  <FormGroup
    tag="div"
  >
    <Label
      for="date-picker"
      tag="label"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      Date*
    </Label>
    <br />
    <withStyles(SingleDatePicker)
      anchorDirection="left"
      appendToBody={false}
      block={false}
      calendarInfoPosition="bottom"
      customCloseIcon={null}
      customInputIcon={null}
      date={"1970-01-01T00:00:00.000Z"}
      daySize={39}
      disableScroll={false}
      disabled={false}
      displayFormat={[Function]}
      enableOutsideDays={false}
      firstDayOfWeek={null}
      focused={false}
      hideKeyboardShortcutsPanel={false}
      horizontalMargin={0}
      horizontalMonthPadding={13}
      id="date-picker"
      initialVisibleMonth={null}
      inputIconPosition="before"
      isDayBlocked={[Function]}
      isDayHighlighted={[Function]}
      isOutsideRange={[Function]}
      isRTL={false}
      keepFocusOnInput={false}
      keepOpenOnDateSelect={false}
      monthFormat="MMMM YYYY"
      navNext={null}
      navPrev={null}
      noBorder={false}
      numberOfMonths={1}
      onClose={[Function]}
      onDateChange={[Function]}
      onFocusChange={[Function]}
      onNextMonthClick={[Function]}
      onPrevMonthClick={[Function]}
      openDirection="down"
      orientation="horizontal"
      phrases={
        Object {
          "calendarLabel": "Calendar",
          "chooseAvailableDate": [Function],
          "clearDate": "Clear Date",
          "closeDatePicker": "Close",
          "dateIsSelected": [Function],
          "dateIsUnavailable": [Function],
          "enterKey": "Enter key",
          "escape": "Escape key",
          "hideKeyboardShortcutsPanel": "Close the shortcuts panel.",
          "homeEnd": "Home and end keys",
          "jumpToNextMonth": "Move forward to switch to the next month.",
          "jumpToPrevMonth": "Move backward to switch to the previous month.",
          "keyboardNavigationInstructions": "Press the down arrow key to interact with the calendar and
  select a date. Press the question mark key to get the keyboard shortcuts for changing dates.",
          "keyboardShortcuts": "Keyboard Shortcuts",
          "leftArrowRightArrow": "Right and left arrow keys",
          "moveFocusByOneDay": "Move backward (left) and forward (right) by one day.",
          "moveFocusByOneMonth": "Switch months.",
          "moveFocusByOneWeek": "Move backward (up) and forward (down) by one week.",
          "moveFocustoStartAndEndOfWeek": "Go to the first or last day of a week.",
          "openThisPanel": "Open this panel.",
          "pageUpPageDown": "page up and page down keys",
          "questionMark": "Question mark",
          "returnFocusToInput": "Return to the date input field.",
          "selectFocusedDate": "Select the date in focus.",
          "showKeyboardShortcutsPanel": "Open the keyboard shortcuts panel.",
          "upArrowDownArrow": "up and down arrow keys",
        }
      }
      placeholder="Date"
      readOnly={false}
      regular={false}
      renderCalendarInfo={null}
      renderDayContents={null}
      renderMonthElement={null}
      renderMonthText={null}
      reopenPickerOnClearDate={false}
      required={false}
      screenReaderInputMessage=""
      showClearDate={false}
      showDefaultInputIcon={false}
      small={false}
      verticalHeight={null}
      verticalSpacing={22}
      weekDayFormat="dd"
      withFullScreenPortal={false}
      withPortal={false}
    />
  </FormGroup>
  <FormGroup
    tag="div"
  >
    <Button
      color="primary"
      tag="button"
    >
      Save Retro
    </Button>
  </FormGroup>
</Form>
`;
